app/src/main/java/de/mvhs/android/zeiterfassung/MainActivity.java

```diff
@@ -16,69 +19,129 @@
-        final EditText endTime = (EditText)findViewById(R.id.EndTime);
-
-        endCommand.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                // Lesen der Datensätze
-                Cursor data = getContentResolver().query(
-                        TimelogContract.Timelog.CONTENT_URI, // URI
-                        null, // Welche Spalten
-                        null, // Filter
-                        null, // Filter Parameter
-                        TimelogContract.Timelog.Columns._ID + " DESC"); // Sortierung
-
-                if (data.moveToFirst()){
-                    long id = data.getLong(data.getColumnIndex(BaseColumns._ID));
-                    String start = data.getString(
-                            data.getColumnIndex(TimelogContract.Timelog.Columns.START));
-
-                    try {
-                        Date startTime = TimelogContract.Converter.DB_DATE_TIME_FORMATTER
-                                .parse(start);
-
-                        start = _UI_DATE_FORMATTER.format(startTime);
-                    } catch (ParseException e) {
-                        e.printStackTrace();
-                    }
-
-                    endTime.setText("ID: " + id + ", Start: "  +start);
-                }
-            }
-        });
+  private Uri recordUriWithEndDateMissing = null;
+  private TimelogEntryModel currentTimelogEntry = null;
+
+  private DateFormat _UI_DATE_FORMATTER = DateFormat.getDateTimeInstance(
+      DateFormat.SHORT,
+      DateFormat.SHORT
+  );
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+
+    TimelogEntryModel.setContentResolver( getContentResolver() );
+
+    final Button startButton = (Button) findViewById( R.id.StartCommand );
+    final Button endButton = (Button) findViewById( R.id.EndCommand );
+    final Cursor emptyEndDateRecord = findRecordWithEmptyEndDate();
+
+    if( emptyEndDateRecord == null ) {
+      startButton.setEnabled( true );
+
+    } else {
+      try {
+        this.currentTimelogEntry = new TimelogEntryModel( emptyEndDateRecord );
+        final EditText startTimeText = (EditText) findViewById( R.id.StartTime );
+        startTimeText.setText( currentTimelogEntry.getUiStartDateString() );
+
+      } catch( ParseException e ) {
+        Log.w( "MainActivity", "Error while creating TimelogEntryModel from record with empty end date: " + String.valueOf( emptyEndDateRecord ));
+      }
+
+      endButton.setEnabled( true );
     }
+  }
+
+  private Cursor findRecordWithEmptyEndDate() {
+    final String whereEndDateIsNull = TimelogContract.Timelog.Columns.END + " IS NULL";
+    final String orderByIdDesc = BaseColumns._ID + " DESC";
+
+    final Cursor records = getContentResolver().query( TimelogContract.Timelog.CONTENT_URI, null, whereEndDateIsNull, null, orderByIdDesc );
+    Cursor returnRecord = null;
+
+    if( records == null || records.getCount() == 0 ) {
```

Comment:
Es geht hier weniger um die Knappheit, als um die Perfomance. Bei einer Liste/Collection, wissen Sie nicht, wie _leicht_ die Anzahl der Elemente bestimmt werden kann. Eventuelle muss über alle Elemente zuerst iterriert werden, bis man die genaue Anzahl kennt. In der C# Welt wird deshalb bei LINQ Operationen (Collections) immer empfohlen statt zum Beispiel `data.Count() > 0` auf `data.Any()` zu Prüfen (in unserem Fall also `data.moveToFirst()`.